piksi_tools/serial_link.py

```diff
@@ -9,447 +9,162 @@
-  item : object
-    Python object to serialize
-
-  Returns
-  ----------
-  (delta, timestamp, item) : tuple
-    delta = msec reference timestamp (int)
-    timestamp = current timestamp (int - UTC epoch)
-    item = Python object to serialize
+  use_ftdi : bool
+    For serial driver, use the pyftdi driver, otherwise use the pyserial driver.
+  port : string
+    Serial port to read.
+  baud : int
+    Serial port baud rate to set.
+  filename : string
+    File to read SBP messages from.
   """
-  timestamp = calendar.timegm(time.gmtime())
-  delta = int((time.time() - t0)*1000)
-  return (delta, timestamp, item)
+  if filename:
+    return FileDriver(filename)
+  if use_ftdi:
+    return PyFTDIDriver(baud)
+  return PySerialDriver(port, baud)
 
-def format_log_json_entry(t0, item):
+def get_logger(use_log, use_json, use_byte, filename):
   """
-  Generates a Python object with logging metadata.
+  Get a logger based on configuration options.
 
   Parameters
   ----------
-  t0 : float
-    Reference time (seconds)
-  item : object
-    Python object to serialize
-
-  Returns
-  ----------
-  {'delta': delta, 'timestamp': timestamp, 'data': data} : dict
-    delta = msec reference timestamp (int)
-    timestamp = current timestamp (int - UTC epoch)
-    data = Python object to JSON serialize
+  use_log : bool
+    Whether to log or not.
+  use_json : bool
+    Log out JSON records, otherwise log out bytes or pickled records.
+  use_byte : bool
+    Log out bytes, otherwise log out pickled records.
+  filename : string
+    File to log to.
   """
-  timestamp = calendar.timegm(time.gmtime())
-  delta = int((time.time() - t0)*1000)
-  data = item.to_json_dict()
-  return {'delta': delta, 'timestamp': timestamp, 'data': data}
-
-class Watchdog:
+  if not use_log:
+    return NullLogger()
+  print "Logging at %s." % filename
+  if use_json:
+    return JSONLogger(filename)
+  if use_byte:
+    return ByteLogger(filename)
+  return PickleLogger(filename)
+
+def printer(sbp_msg):
   """
-  Watchdog wraps a timer with a callback that can rearm the timer.
+  Default print callback
 
   Parameters
   ----------
-  timeout : float
-    timeout of timer in seconds
-  alarm : callback
-    function to call when/if timer expires
+  sbp_msg: SBP
+    SBP Message to print out.
   """
-  def __init__(self, timeout, alarm):
-    self.timeout = timeout
-    self.alarm = alarm
-    self.timer = None
-
-  def __call__(self, *args):
-    self.call()
-
-  def call(self):
-    """
-    Rearm the timer.
-    """
-    if self.timer:
-      self.timer.cancel()
-    self.timer = threading.Timer(self.timeout, self.alarm)
-    self.timer.daemon = True
-    self.timer.start()
+  sys.stdout.write(sbp_msg.payload)
 
-def default_watchdog_alarm():
+def main():
   """
-  Called when the watchdog timer alarms. Will raise a KeyboardInterrupt to the
-  main thread and exit the process.
+  Get configuration, get driver, get logger, and build handler and start it.
```

Comment:
so the actual serial link command line tool's going to continue to live here instead of at: https://github.com/swift-nav/libsbp/blob/master/python/sbp/client/main.py. In that case, we'll probably have to clarify with "libsbp has the dev libraries" and "piksi_tools has the actual command line tool" or something.